app/presenters/coronavirus/sub_section_json_presenter.rb

```diff
@@ -34,7 +34,25 @@ def output
   end
 
   def sub_sections
-    content_groups.map { |content_group| sub_section_hash_from_content_group(content_group) }
+    content = content_groups.map { |content_group| sub_section_hash_from_content_group(content_group) }
+    content.unshift build_action_link if sub_section.action_link_present?
```

Comment:
Have updated `sub_sections` so that the action_link is added first.  We still need the guard clauses to stop `nil` or `{}` being created, which would mean some existing tests would fail.  But, hopefully it simplifies things a little and makes the code more readable.  I've also made `build_action_link` method private and removed the tests that call it directly.

Reply:
You shouldn't need the guard because of the compact. if action_link returns nil and then you compact the array it'll be an empty array.

Reply:
Have removed the need for the guard.